redash/query_runner/__init__.py

```diff
@@ -208,6 +219,22 @@ def _get_tables_stats(self, tables_dict):
                 res = self._run_query_internal("select count(*) as cnt from %s" % t)
                 tables_dict[t]["size"] = res[0]["cnt"]
 
+    @property
+    def supports_auto_limit(self):
+        return True
+
+    def apply_auto_limit(self, query_text, set_auto_limit):
+        if set_auto_limit:
+            from redash.query_runner.databricks import split_sql_statements
+            query_list = split_sql_statements(query_text)
+            # we only check for last one in the list because it is the one that we show result
+            last_query = query_list[-1]
+            if query_is_select_no_limit(last_query):
+                query_list[-1] = add_limit_to_query(last_query)
+            return ";\n".join(query_list)
```

Comment:
This is a leaky abstraction. The knowledge of splitting is inside `split_sql_statements`, but gluing them back together is here.